pkg/kubelet/lifecycle/handlers.go

```diff
@@ -149,6 +152,10 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 
 			// clear err since the fallback succeeded
 			if httpErr == nil {
+				if eventRecorder != nil {
+					// report the fallback with an event
+					eventRecorder.Event(pod, v1.EventTypeWarning, "LifecycleHTTPFallback", fmt.Sprintf("request to HTTPS lifecycle hook %s got HTTP response, retry with HTTP succeeded", req.URL.Host))
```

Comment:
Do we know whether this is the PostStart or PreStop handler at this point? That might be useful to include in the event if so.

Reply:
we don't, but the port number should be sufficient to identify the problematic endpoint